src/preprocessing/passes/global_negate.cpp

```diff
@@ -90,21 +92,30 @@ void GlobalNegate::simplify(std::vector<Node>& assertions)
   Trace("cbqi-gn-debug") << "...got (pre-rewrite) : " << body << std::endl;
   body = Rewriter::rewrite(body);
   Trace("cbqi-gn") << "...got (post-rewrite) : " << body << std::endl;
+  return body;
+}
+}  // namespace
+
+GlobalNegate::GlobalNegate(PreprocessingPassContext* preprocContext)
+    : PreprocessingPass(preprocContext, "global-negate"){};
 
-  Node truen = nm->mkConst(true);
-  for (unsigned i = 0, size = assertions.size(); i < size; i++)
+PreprocessingPassResult GlobalNegate::applyInternal(
+    AssertionPipeline* assertionsToPreprocess)
+{
+  if (assertionsToPreprocess->size() > 0)
   {
-    if (i == 0)
-    {
-      assertions[i] = body;
-    }
-    else
+    NodeManager* nm = NodeManager::currentNM();
+    assertionsToPreprocess->replace(
```

Comment:
I'm not sure that we should use `replace` here, or follow the original behavior of replacing assertions by `true`.  I actually think its more appropriate to call:
```
Node ret = simplify(assertionsToPreprocess->ref(), nm);
assertionsToPreprocess->clear();
assertionsToPreprocess->push_back(ret);
```

Perhaps @4tXJ7f can comment?

Reply:
As far as I know, we have to keep the number of assertions the same or larger, otherwise a bunch of other things will break (I don't remember what concretely but one issue is for example that we'd have to update `d_realAssertionsEnd`). In the future, it would be better to be able to actually remove assertions, yes.